communication/src/main/java/org/jboss/da/communcation/pnc/authentication/PNCAuthFilter.java

```diff
@@ -0,0 +1,25 @@
+package org.jboss.da.communcation.pnc.authentication;
+
+import org.jboss.da.communcation.pnc.authentication.PNCAuthentication;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import java.io.IOException;
+
+/**
+ * Filter to add authorization header to all http requests to PNC
+ */
+public class PNCAuthFilter implements ClientRequestFilter {
+
+    /**
+     * It works by add the Authorization key, followed by "Bearer <token>"
+     * to the http request
+     *
+     * @param requestContext
+     * @throws IOException
+     */
+    public void filter(ClientRequestContext requestContext) throws IOException {
+        String token = PNCAuthentication.authenticate();
```

Comment:
It depends on the server configuration for tokens validity. As default Keycloak tokens are valid for 5 minutes. It is possible for sure to reuse the access_token, but then you would need to implement the logic to be prepared getting from REST service endpoint response as: "access token expired" and then re-authenticate and try again to access REST service. Of course, refresh tokens are also possibility for getting fresh access_tokens, but it would require enen more additional coding to handle the refreshing tokens.